File: python/muse/algorithms/number_theory/parity.py
def modulo_is_even(n: int) -> bool:
    return n % 2 == 0


def modulo_is_odd(n: int) -> bool:
    return n % 2 != 0


def bitwise_is_even(n: int) -> bool:
    return (n & 1) == 0


def bitwise_is_odd(n: int) -> bool:
    return (n & 1) != 0

File: rust/muse/src/algorithms/number_theory/parity.rs
pub fn modulo_is_even(n: i64) -> bool {
    n % 2 == 0
}

pub fn modulo_is_odd(n: i64) -> bool {
    n % 2 != 0
}

pub fn bitwise_is_even(n: i64) -> bool {
    (n & 1) == 0
}

pub fn bitwise_is_odd(n: i64) -> bool {
    (n & 1) != 0
}

#[cfg(test)]
pub(crate) mod tests;
